### PR TITLE
workflows: add Gosec workflow

### PR DESCRIPTION
**Pull request by AdamKorcz**
Adds a workflow for the Gosec static analysis tool and ignores false positive findings.

### CODE REVIEW

**Review by priteshbandi — APPROVED** [review]
LGTM

File: .github/workflows/gosec.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    branches:
+      - main
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
```
Comment (priteshbandi) [inline .github/workflows/gosec.yml:16]:
```suggestion
        uses: actions/checkout@v3
```

**Review by priteshbandi — APPROVED** [review]
LGTM

File: .github/workflows/gosec.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v3
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
```
Comment (shizhMSFT) [inline .github/workflows/gosec.yml:18]: Can we use a fixed version?

**Pull request closed without merging**